src/test/java/bio/terra/common/auth/AuthService.java

```diff
@@ -27,66 +22,57 @@
     private List<String> directAccessScopes = Arrays.asList(
         "https://www.googleapis.com/auth/bigquery",
         "https://www.googleapis.com/auth/devstorage.full_control");
-    private NetHttpTransport httpTransport;
-    private JacksonFactory jsonFactory = JacksonFactory.getDefaultInstance();
-    private File pemfile;
     private String saEmail;
-    private Map<String, String> userTokens = new HashMap<>();
-    private Map<String, String> directAccessTokens = new HashMap<>();
     private TestConfiguration testConfig;
 
+    private Map<String, GoogleCredentials> userCreds = new HashMap<>();
+    private Map<String, GoogleCredentials> directAccessCreds = new HashMap<>();
+
     @Autowired
     public AuthService(TestConfiguration testConfig) throws Exception {
         this.testConfig = testConfig;
-        Optional<String> pemfilename = Optional.ofNullable(testConfig.getJadePemFileName());
-        pemfilename.ifPresent(s -> pemfile = new File(s));
         saEmail = testConfig.getJadeEmail();
-        httpTransport = GoogleNetHttpTransport.newTrustedTransport();
     }
 
     public String getAuthToken(String userEmail) {
-        if (!userTokens.containsKey(userEmail)) {
-            userTokens.put(userEmail, makeToken(userEmail));
-        }
-        return userTokens.get(userEmail);
+        return getCredentials(userEmail).getAccessToken().getTokenValue();
     }
 
     public String getDirectAccessAuthToken(String userEmail) {
-        if (!directAccessTokens.containsKey(userEmail)) {
-            directAccessTokens.put(userEmail, makeDirectAccessToken(userEmail));
+        if (!directAccessCreds.containsKey(userEmail)) {
+            directAccessCreds.put(userEmail, makeDirectAccessCredentials(userEmail));
         }
-        return directAccessTokens.get(userEmail);
+        return directAccessCreds.get(userEmail).getAccessToken().getTokenValue();
     }
 
-    private GoogleCredential buildCredential(String email, List<String> scopes)
-            throws IOException, GeneralSecurityException {
-        if (!Optional.ofNullable(pemfile).isPresent()) {
-            throw new IllegalStateException(String.format("pemfile not found: %s", testConfig.getJadePemFileName()));
+    public GoogleCredentials getCredentials(String userEmail) {
+        if (!userCreds.containsKey(userEmail)) {
+            userCreds.put(userEmail, makeToken(userEmail));
         }
-        return new GoogleCredential.Builder()
-            .setTransport(httpTransport)
-            .setJsonFactory(jsonFactory)
-            .setServiceAccountId(saEmail)
-            .setServiceAccountPrivateKeyFromPemFile(pemfile)
-            .setServiceAccountScopes(scopes)
-            .setServiceAccountUser(email)
-            .build();
+        return userCreds.get(userEmail);
     }
 
-    private String makeDirectAccessToken(String userEmail) {
+    private GoogleCredentials buildCredentials(String email, List<String> scopes) throws IOException {
```

Comment:
This looks like it only used in one place. Do we really need another method?

Reply:
Probably not, I'm just going to revert this file and make a more localized change.